Lua cvar library

viviicat opened the pull request.

viviicat committed:
cvars!

viviicat committed:
Color test

gheshu commented on src/scriptsys/scr_cvar.c:
```diff
@@ -0,0 +1,161 @@
+
+static void scr_push_vec(lua_State* L, float4 vec)
+{
+	lua_createtable(L, 4, 0);
```
table == {} or an object in lua, right?
A TODO for later: we should figure out a faster way to pass structs to lua, maybe via FFI, or maybe just push 4 floats?
You may want to use VEC_CALL on anything taking or returning a math type like float4 (unless its passed as a function pointer to lua).

viviicat replied:
I think we could use `UserData` for this, and then have the X, Y, Z, W be fetched using metatables. Would have to profile performance!

viviicat replied:
#23

gheshu commented on src/scriptsys/scr_cvar.h:
```diff
@@ -0,0 +1,8 @@
+
+PIM_C_BEGIN
+
+void scr_cvar_init(lua_State* L);
```
Would be good to have a matching shutdown for any init function, so that we can hot reload sections of the engine.

Vethanis reviewed (approved):
Cool!

viviicat committed:
Add VEC_CALL

viviicat committed:
Add shutdown stubs.

At the moment everything gets cleaned up when we clean up the lua_State, so we don't need to do anything else to be fully cleaned.

viviicat committed:
Missing...

viviicat committed:
Update init.lua to use cvar system.
